### ISSUE TITLE
Modify the `Model` API

### ISSUE THREAD

**Issue by delhomer** [Oslandia/mapillary_classification#35]
Adapt the `Model` API so as to make it closer from popular machine learning libraries. Consider these methods:
- fit : train model
- predict : do inference on test data
- save : serialize model
- score : give main metric(s)

**Comment by garaud** [comment]
I think we should get a similar API to the Keras "application" models, i.e. https://keras.io/applications/
An example of DenseNet https://github.com/keras-team/keras/blob/master/keras/applications/densenet.py There are some functions which build the network.

**Comment by delhomer** [comment]
Actually we will certainly use the Keras `Model`, as we moved the network architecture creation to a specific `ConvolutionalNeuralNetwork` class. The native API already has all the sought methods. In this way, the issue seems quite irrelevant, I close it.
